scipy/optimize/_basinhopping.py

```diff
@@ -152,7 +152,7 @@ def one_cycle(self):
 
         accept, minres = self._monte_carlo_step()
 
-        if accept:
+        if accept and minres.success is not scipy.optimize.InvalidResult:
```

Comment:
This will still print `accept 1` when it hits `self.print_report(minres.fun, accept)`, even though the step is not "accepted".  Is that good or bad?  

Adding the `success` criteria to the `accept_tests` isn't possible because those are based only on the new and old values.